omigami/feature_selector.py

```diff
@@ -22,68 +37,198 @@ def __init__(
+        input_data: InputDataset,
+        outer_split: Split,
+        data_splitter: DataSplitter,
+    ) -> OuterLoopResults:
+        min_eval, mid_eval, max_eval = self.evaluate_min_mid_and_max_features(
+            input_data,
+            feature_elimination_results.best_features,
+            outer_split,
+            data_splitter,
+        )
+        outer_loop_results = OuterLoopResults(
+            min_eval=min_eval,
+            mid_eval=mid_eval,
+            max_eval=max_eval,
+            n_features_to_score_map=feature_elimination_results.n_features_to_score_map,
         )
+        return outer_loop_results
+
+    def evaluate_min_mid_and_max_features(
+        self,
+        input_data: InputDataset,
+        best_features: SelectedFeatures,
+        split: Split,
+        data_splitter: DataSplitter,
+    ) -> Tuple[
+        FeatureEvaluationResults, FeatureEvaluationResults, FeatureEvaluationResults
+    ]:
+        min_feats = best_features.min_feats
+        mid_feats = best_features.mid_feats
+        max_feats = best_features.max_feats
+
+        data_min_feats = data_splitter.split_data(input_data, split, min_feats)
+        data_mid_feats = data_splitter.split_data(input_data, split, mid_feats)
+        data_max_feats = data_splitter.split_data(input_data, split, max_feats)
+
+        min_eval = self.feature_evaluator.evaluate_features(data_min_feats, min_feats)
+        mid_eval = self.feature_evaluator.evaluate_features(data_mid_feats, mid_feats)
+        max_eval = self.feature_evaluator.evaluate_features(data_max_feats, max_feats)
+
+        return min_eval, mid_eval, max_eval
+
+    def _select_best_features(
+        self, repetition_results: FeatureSelectionResults
+    ) -> SelectedFeatures:
+        self.results = self.post_processor.fetch_results(repetition_results)
+        selected_features = self.post_processor.select_features(self.results)
+        return selected_features
 
     def get_validation_curves(self) -> Dict[str, List]:
-        return self.post_processor.get_validation_curves(self._results)
+        return self.post_processor.get_validation_curves(self.results)
+
+    def get_selected_features(self, feature_names: List[str] = None):
+
+        if not self.is_fit:
+            # TODO: custom exception
+            raise RuntimeError("The feature selector is not fit yet")
+
+        if feature_names is not None:
+            min_names = [feature_names[f] for f in self._selected_features.min_feats]
+            mid_names = [feature_names[f] for f in self._selected_features.mid_feats]
+            max_names = [feature_names[f] for f in self._selected_features.max_feats]
+
+            return SelectedFeatures(
+                min_feats=min_names, max_feats=max_names, mid_feats=mid_names,
+            )
+
+        # TODO: in one case we return a "copy" in this case we return a reference.
+        # maybe it's better to always return a copy
```

Comment:
Copy is better if the user changes something and it is a reference it will mess with this classes state unexpectedly.